packages/amplication-client/src/Workspaces/hooks/usePendingChanges.ts

```diff
@@ -28,8 +28,7 @@ const usePendingChanges = (currentProject: models.Project | undefined) => {
   });
 
   useEffect(() => {
-    if (!pendingChangesData || !pendingChangesData.pendingChanges.length)
-      return;
+    if (!pendingChangesData) return;
```

Comment:
instead of changing this condition and rerender the component, i suggest we add another useEffect that check the changes of currentProject (with the url project param) and accordingly refetch pending changes

Reply:
Sounds good to me. I am on it 🙌

Reply:
@barshimi I went through the logic once again, and I am not sure if refetching from another userEffect will be enough.
Because this specific block of code is responsible for setting the changes in the state.

Even if we refetch pending changes elsewhere we might still have to write the same code about setting state again, isn' it?